src/azul/service/hca_response_v5.py

```diff
@@ -362,6 +362,28 @@ def assert_no_duplicate_value(key):
             for file in files
         ]
 
+        # Check each stratum for dimension values that contain comma-separated
```

Comment:
This is ok but a little pedestrian. There is value in looking things a bit more abstractly and use a functional style. It leads to code that is more less likely to have bugs and more efficient. It can be harder to understand but only to reader unfamiliar with the idiom. The expansion of the comma-separated values is a very straight-forward combinatorial operation: a product.

The comment that I added in the code section above this one didn't mention the commas and must be updated, too.